### ISSUE TITLE
IdS4 problem with Authority URL

### ISSUE THREAD

**Issue by nicolasr75** [IdentityServer/IdentityServer4#1577]
I am connecting with oidc-client.js to an ASP.NET Core Web Api which I want to protect with IdS4/AccessTokenValidation. The api and the IdS run on the same server on port 443. The server is behind a router/firewall and I have configured a port forwarding for port 443.

The problem is this:
when I configure the authority URL in the api to https://localhost... this results in "Issuer name does not match authority" because the authority in the oidc-client is of course set to the WAN address.
But when I configure the authority URL to the WAN address, my router blocks the request to the IdS. I think this is because there is no NAT loopback.

Is this a known problem? Do I misunderstand anything? Or is NAT loopback a required feature for this scenario? Unfortunately the router I am working with doesn't seem to support it.

**Comment by brockallen** [comment]
Have you looked at the load balancing and deployment docs?

**Comment by nicolasr75** [comment]
Sorry for my ignorance.
You mean this?
http://docs.identityserver.io/en/release/topics/deployment.html
I don't see much about IdentityServer that could help me except maybe the given link to MS
https://github.com/aspnet/Docs/issues/2384

Given your hint about load balancing I found his article which seems very helpful:
http://amilspage.com/set-identityserver4-url-behind-loadbalancer/

As a quick fix I solved the problem using the IssuerUri option.
http://docs.identityserver.io/en/release/reference/options.html
(Although the reference recommends to not use it)
